src/telemetry/connectionTelemetry.ts

```diff
@@ -110,15 +108,22 @@ export async function getConnectionTelemetryProperties(
     const username = connectionString.username ? 'DEFAULT' : 'NONE';
     const authStrategy = authMechanism ?? username;
 
+    const hostname = await getHostnameForConnection(
+      dataService.getConnectionString()
```

Comment:
Looks like you have `const connectionString = dataService.getConnectionString();` on line `106` already available